go/vt/vttablet/grpctmclient/client.go

```diff
@@ -185,7 +186,19 @@ func (client *grpcClient) dialPool(ctx context.Context, tablet *topodatapb.Table
 
 	result := <-c
 	c <- result
-	return result.client, nil
+	invalidator := func(err error) {
+		if err == nil {
+			return
+		}
+		if _, ok := status.FromError(err); !ok {
+			// Not a gRPC error
+			return
+		}
+		client.mu.Lock()
+		defer client.mu.Unlock()
+		delete(client.rpcClientMap, addr)
```

Comment:
We should go over the map and close all the grpc connections - 
```go
for tm := range client.rpcClientMap[addr] {
	tm.cc.Close()
}
```

Reply:
Done.